src/NuGet.Clients/NuGet.PackageManagement.UI/Resources/Resources.xaml.cs

```diff
@@ -13,7 +17,13 @@ public SharedResources()
         private void PackageIconImage_ImageFailed(object sender, ExceptionRoutedEventArgs e)
         {
             var image = sender as Image;
-            image.Source = Images.DefaultPackageIcon;
+
+            e.Handled = true; // don't repropagate the event
```

Comment:
How is the default item set now?

Reply:
Via the IconUrlImageCacheConverter. If the image failed to load/process, then, in the cache, there should be an entry with the default image [(ref. 1)]() [(ref. 2)](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs#L131).

Also, the [Image control ](https://github.com/NuGet/NuGet.Client/blob/8f106e41e7d73ff76432955dd85e966d18bddede/src/NuGet.Clients/NuGet.PackageManagement.UI/Resources/Resources.xaml#L696)puts the default icon in null or failed events

Reply:
Thanks. 

I think a nicer explanation in the original PR body would've been helpful.